### PR TITLE
Stand up build signing in taskgraph

### PR DESCRIPTION
**Pull request by ahal**
This depends on @sarah-clements 's PR #5145 . I'm submitting now for testing purposes (as `staging-focus-android` doesn't have necessary scopes yet).

### CODE REVIEW

File: taskcluster/ci/ui-test/kind.yml
Comment (sarah-clements) [inline taskcluster/ci/ui-test/kind.yml]: Once you add the signing kind, I think you'll see why the the multi_dep transform is needed, unless there's some other way to have one task pending until another task completes. Also, if you look at fenix, the dependency should change once you add the signing kind; this task will no longer be primarily dependent on the build to finish (that will be the signing kind), and this in turn will depend on the signing kind to finish. The reason why I didn't set this kind up is because for the first pr, the `focusDebug` is signed by default with a debug key, and that's all that's needed for the ui-test kind.

File: .taskcluster.yml
Comment (escapewindow) [inline .taskcluster.yml:121]: Removing the two lines above may fix the CoT issues. I think we use `checks` instead of `statuses` now.

Comment (escapewindow) [reply .taskcluster.yml:121]: Ben adds `adding https://github.com/mozilla-mobile/fenix/blob/main/.taskcluster.yml#L3 might be good too, regardless`

`verify_cot` was useful in debugging here; it may be a good tool to start using.

File: taskcluster/focus_android_taskgraph/transforms/signing.py
Comment (bhearsum) [inline taskcluster/focus_android_taskgraph/transforms/signing.py:58]: I'm a little bit surprised we're signing test builds with real certs, but Fenix is already doing it, so there's probably a good reason....

File: taskcluster/focus_android_taskgraph/transforms/signing.py
Comment (bhearsum) [inline taskcluster/focus_android_taskgraph/transforms/signing.py:71]: Typo here: s/debut/debug/

**Review by bhearsum — APPROVED** [review]
This seems like it should be good, modulo the one typo below!

File: taskcluster/focus_android_taskgraph/transforms/signing.py
Comment (ahal) [reply taskcluster/focus_android_taskgraph/transforms/signing.py:58]: I don't know.. this was just copy / pasted over from Fenix. Might be better to start without it and then add it later if it turns our it's necessary? Also I guess we can remove these `mozillaonline` values?

File: taskcluster/focus_android_taskgraph/transforms/signing.py
Comment (ahal) [reply taskcluster/focus_android_taskgraph/transforms/signing.py:71]: Good catch, this typo is actually in `fenix` too...

File: taskcluster/focus_android_taskgraph/transforms/signing.py
Comment (ahal) [reply taskcluster/focus_android_taskgraph/transforms/signing.py:58]: On closer look, none of these (except `nightly`) are even valid build types in `focus`.. I'll remove them.

Comment (bhearsum) [reply taskcluster/focus_android_taskgraph/transforms/signing.py:58]: Yes, actually - the mozillaonline ones can defiintely be removed...

**Comment by ahal** [comment]
I dug into why some of the signed arfitacts have `unsigned` in their name. Turns out it's just because that's what the build apk artifacts have as well. The build artifact names in turn come from the command `./gradlew --no-daemon --quiet printVariants`. As far as I can tell there's no mechanism for `signingscript` to change the name of the artifact being signed.. so that means if folks want the `unsigned` gone, it will need to be adjusted in the gradle configuration. That is not something I know how to do, so I'll leave it up to the focus team to make whatever tweaks they would like there as a follow-up to this PR :).

This PR is ready to go and just waiting on #5145 to merge.

**Comment by escapewindow** [comment]
+1, I recomment `target.apk` as a name since we've standardized on it elsewhere.

**Comment by ahal** [comment]
I've made a fair amount of changes since last time, so re-requesting review.

File: taskcluster/ci/signing/kind.yml
Comment (bhearsum) [inline taskcluster/ci/signing/kind.yml:31]: Eventually we'll want `github-release` here too, I think? It can probably be added now without causing any issuse, if you want. Definitely not blocking though.

**Review by bhearsum — APPROVED** [review]

**Comment by ahal** [comment]
I think I saw that the ui-tests were removed from mergify's check, so hopefully this lands automatically once other tasks are green.

**Pull request merged**
